synaptic_reconstruction/inference/mitochondria.py

```diff
@@ -0,0 +1,139 @@
+import time
+from typing import Dict, List, Optional, Tuple, Union
+import torch
+import torch_em
+import elf.parallel as parallel
+import numpy as np
+import xarray
+from tqdm import tqdm
+
+from skimage.transform import rescale, resize
+from skimage.measure import label, regionprops
+from skimage.morphology import binary_closing, remove_small_holes
+
+DEFAULT_TILING = {
+    "tile": {"x": 512, "y": 512, "z": 64},
+    "halo": {"x": 64, "y": 64, "z": 8},
+}
+
+
+def _run_segmentation(
+    foreground, boundaries, verbose, min_size,
+    # blocking shapes for parallel computation
+    block_shape=(128, 256, 256),
+    halo=(48, 48, 48)
+):
+
+    # get the segmentation via seeded watershed
+    t0 = time.time()
+    seeds = parallel.label((foreground - boundaries) > 0.5, block_shape=block_shape, verbose=verbose)
+    if verbose:
+        print("Compute connected components in", time.time() - t0, "s")
+
+    t0 = time.time()
+    dist = parallel.distance_transform(seeds == 0, halo=halo, verbose=verbose, block_shape=block_shape)
+    if verbose:
+        print("Compute distance transform in", time.time() - t0, "s")
+
+    t0 = time.time()
+    mask = (foreground + boundaries) > 0.5
+    seg = np.zeros_like(seeds)
+    seg = parallel.seeded_watershed(
+        dist, seeds, block_shape=block_shape,
+        out=seg, mask=mask, verbose=verbose, halo=halo,
+    )
+    if verbose:
+        print("Compute watershed in", time.time() - t0, "s")
+
+    t0 = time.time()
+    ids, sizes = parallel.unique(seg, return_counts=True, block_shape=block_shape, verbose=verbose)
+    filter_ids = ids[sizes < min_size]
+    seg[np.isin(seg, filter_ids)] = 0
+    if verbose:
+        print("Size filter in", time.time() - t0, "s")
+    return seg
+
+
+def _get_prediction_torch_em(
```

Comment:
Let's refactor this into a new `util.py` file and remove the `_` at the beginning. Then we can reuse it in other segmentation functions.